### ISSUE TITLE
New rust core PR breaks rocket_codegen on nightly

### ISSUE THREAD

**Issue by xorl** [SergioBenitez/Rocket#503]
https://github.com/rust-lang/rust/pull/46550 breaks rocket_codegen, global is now PathKind::{Global, Std, Local}

Tried to fix it locally but failed to create a working PR.

This breaks on 0.3.3. from the current nightly. 

```
   Compiling rocket_codegen v0.3.3
error[E0560]: struct `syntax_ext::deriving::generic::ty::Path<'_>` has no field named `global`
  --> /Users/rlewon/.cargo/registry/src/github.com-1ecc6299db9ec823/rocket_codegen-0.3.3/src/decorators/derive_form.rs:76:13
   |
76 |             global: true,
   |             ^^^^^^^ `syntax_ext::deriving::generic::ty::Path<'_>` does not have this field

error[E0560]: struct `syntax_ext::deriving::generic::ty::Path<'_>` has no field named `global`
  --> /Users/rlewon/.cargo/registry/src/github.com-1ecc6299db9ec823/rocket_codegen-0.3.3/src/decorators/derive_form.rs:91:29
   |
91 |                             global: true
   |                             ^^^^^^^ `syntax_ext::deriving::generic::ty::Path<'_>` does not have this field

error[E0560]: struct `syntax_ext::deriving::generic::ty::Path<'_>` has no field named `global`
  --> /Users/rlewon/.cargo/registry/src/github.com-1ecc6299db9ec823/rocket_codegen-0.3.3/src/decorators/derive_form.rs:99:25
   |
99 |                         global: false,
   |                         ^^^^^^^ `syntax_ext::deriving::generic::ty::Path<'_>` does not have this field

error[E0560]: struct `syntax_ext::deriving::generic::ty::Path<'_>` has no field named `global`
   --> /Users/rlewon/.cargo/registry/src/github.com-1ecc6299db9ec823/rocket_codegen-0.3.3/src/decorators/derive_form.rs:109:21
    |
109 |                     global: true,
    |                     ^^^^^^^ `syntax_ext::deriving::generic::ty::Path<'_>` does not have this field

error: aborting due to 4 previous errors

error: Could not compile `rocket_codegen`.
```

**Comment by SergioBenitez** [comment]
This has already been fixed in https://github.com/SergioBenitez/Rocket/commit/4ea9b046cd8cd1f204f7faeb80128a8321c6c461. A new release will be made soon containing this commit.

**Comment by SergioBenitez** [comment]
Released as v0.3.4.

**Comment by xorl** [comment]
Awesome!
